sample-input/gradients/one-directional/one-directional-cmfd.py

```diff
@@ -0,0 +1,83 @@
+import numpy
+from openmoc import *
+import openmoc.log as log
+import openmoc.plotter as plotter
+from openmoc.options import Options
+
+###############################################################################
+#######################   Main Simulation Parameters   ########################
+###############################################################################
+
+options = Options()
+
+num_threads = options.getNumThreads()
+track_spacing = options.getTrackSpacing()
+num_azim = options.getNumAzimAngles()
+tolerance = options.getTolerance()
+max_iters = options.getMaxIterations()
+
+log.set_log_level('NORMAL')
+log.py_printf('TITLE', 'Simulating a one group homogeneous, one directional'
+    ' gradient...')
+
+###############################################################################
+###########################   Creating CMFD Mesh    ###########################
+###############################################################################
+log.py_printf('NORMAL', 'Creating Cmfd mesh...')
+
+cmfd = Cmfd()
+cmfd.setMOCRelaxationFactor(0.6)
+cmfd.setSORRelaxationFactor(1.5)
+cmfd.setLatticeStructure(51,1)
+cmfd.setGroupStructure([1])
+
+###############################################################################
+#########################   Load the Cubic Geometry   #########################
+###############################################################################
+
+log.py_printf('NORMAL', 'Importing cubic geometry...')
+
+import sys
+sys.path.append('..')
+import boundaries
+boundaries.left_bc = VACUUM
+boundaries.right_bc = REFLECTIVE
+boundaries.top_bc = REFLECTIVE
+boundaries.bottom_bc = REFLECTIVE
+from cube import *
+
+geometry.setCmfd(cmfd)
+geometry.initializeFlatSourceRegions()
```

Comment:
Actually it is needed in the `one-directional-cmfd.py` since `geometry.initializeFlatSourceRegions()` initializes CMFD, which sets the boundary conditions for CMFD. Thus if the BCs change, cmfd needs to be reinitialized by the geometry or the appropriate BC in CMFD needs to be set by the user. I would recommend keeping as is in this file, but removing in the other files without CMFD.

With this in mind, we might consider having `initializeCmfd()` method be called from the `TrackGenerator::generateTracks()` method instead of in `Geometry::initializeFlatSourceRegions` just prior to track generation to avoid any issues of differing BCs. If others agree, this can be added as an issue for a later PR.